uport-defaults/src/main/java/me/uport/sdk/uportdefaults/Extensions.kt

```diff
@@ -0,0 +1,81 @@
+package me.uport.sdk.uportdefaults
+
+import me.uport.sdk.core.EthNetwork
+import me.uport.sdk.ethrdid.EthrDIDResolver
+import me.uport.sdk.httpsdid.WebDIDResolver
+import me.uport.sdk.jsonrpc.JsonRPC
+import me.uport.sdk.universaldid.DIDResolver
+import me.uport.sdk.uportdid.UportDIDResolver
+
+/**
+ *
+ * An extension function used to configure default [DIDResolver] for
+ * all known networks and add using the DIDResolver Builder
+ *
+ */
+ fun DIDResolver.configureDefaultsWithInfura(infuraProjectId: String): DIDResolver {
+
+    val mainnet = EthNetwork(
+        name = "mainnet",
+        networkId = "0x1",
+        rpcUrl = "https://mainnet.infura.io/v3/${infuraProjectId}",
+        ethrDidRegistry = "0xdca7ef03e98e0dc2b855be647c39abe984fcf21b",
+        uPortRegistry = "2ngV6QowStW3ebKXYUhy43wCqeLXcuTfHj2"
+    )
+
+    val rinkeby = EthNetwork(
+        name = "rinkeby",
+        networkId = "0x4",
+        rpcUrl = "https://rinkeby.infura.io/v3/${infuraProjectId}",
+        ethrDidRegistry = "0xdca7ef03e98e0dc2b855be647c39abe984fcf21b",
+        uPortRegistry = "2ogxWTKKfy6kwfqdgdEE6GCdoFD4vm4YRZe"
+    )
+
+    val ropsten = EthNetwork(
+        name = "ropsten",
+        networkId = "0x3",
+        rpcUrl = "https://ropsten.infura.io/v3/${infuraProjectId}",
+        ethrDidRegistry = "0xdca7ef03e98e0dc2b855be647c39abe984fcf21b",
+        uPortRegistry = "2oKVhUttUcwaFAopRBGA21NDJoYcBb3a6iz"
+    )
+
+    val kovan = EthNetwork(
+        name = "kovan",
+        networkId = "0x2a",
+        rpcUrl = "https://kovan.infura.io/v3/${infuraProjectId}",
+        ethrDidRegistry = "0xdca7ef03e98e0dc2b855be647c39abe984fcf21b",
+        uPortRegistry = "354S1QuCzkmKoQ3ADSLp1KtqAe8gZ74F9am"
+    )
+
+    return DIDResolver.Builder()
+        .addResolver(
+            // register default Ethr DID resolver for all known networks
+            EthrDIDResolver.Builder()
+                .addNetwork(mainnet)
+                .addNetwork(rinkeby)
+                .addNetwork(ropsten)
+                .addNetwork(kovan)
+                .build()
+        )
+        .addResolver(
```

Comment:
The reason `UportDIDResolver` takes an `rpc` instance is for mocking.
Normally, `UportDIDResolver` should be able to go to multiple networks, since the network is embedded in the MNID.

We can fix this aspect in a separate PR. With that fix we would only need one instance of UportDIDResolver instead of one for each network.


However, uport-did is deprecated so it's unlikely we'll dedicate more resources there.

Reply:
If we don't want to invest time to allow multi next implementation for `UportDIDResolver` do we leave as is here with multiple instances? Or do we take it out completely?

Reply:
I'm leaning toward taking it out.